src/main/java/seedu/address/logic/parser/gradetrackerparsers/AddAssignmentParser.java

```diff
@@ -16,29 +16,54 @@
 import seedu.address.logic.parser.ParserUtil;
 import seedu.address.logic.parser.Prefix;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.module.ModuleName;
 import seedu.address.model.module.grade.Assignment;
+import seedu.address.model.module.grade.AssignmentName;
+import seedu.address.model.module.grade.AssignmentPercentage;
+import seedu.address.model.module.grade.AssignmentResult;
 
 public class AddAssignmentParser implements Parser<AddAssignmentCommand> {
     @Override
     public AddAssignmentCommand parse(String args) throws ParseException {
         requireNonNull(args);
         Assignment newAssignment;
-        String moduleName;
-        ArgumentTokenizer tokenizer = new ArgumentTokenizer(args, PREFIX_NAME, PREFIX_ADD_NEW_ASSIGNMENT,
-                PREFIX_PERCENTAGE_ASSIGNMENT, PREFIX_RESULT_ASSIGNMENT);
+        AssignmentName assignmentName;
+        AssignmentPercentage assignmentPercentage;
+        AssignmentResult assignmentResult;
+        ModuleName moduleName;
+        ArgumentTokenizer tokenizer = new ArgumentTokenizer(args, PREFIX_NAME, PREFIX_ASSIGNMENT_NAME,
+                PREFIX_ASSIGNMENT_PERCENTAGE, PREFIX_ASSIGNMENT_RESULT);
         ArgumentMultimap argMultimap = tokenizer.tokenize();
-        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADD_NEW_ASSIGNMENT,
-                PREFIX_PERCENTAGE_ASSIGNMENT, PREFIX_RESULT_ASSIGNMENT)
-                || !argMultimap.getPreamble().isEmpty()) {
+        if ((!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ASSIGNMENT_NAME,
+                PREFIX_ASSIGNMENT_PERCENTAGE, PREFIX_ASSIGNMENT_RESULT)
+                || !argMultimap.getPreamble().isEmpty()) && (argMultimap.getValue(PREFIX_ASSIGNMENT_NAME).isPresent()
+                        && argMultimap.getValue(PREFIX_ASSIGNMENT_PERCENTAGE).isPresent()
+                        && argMultimap.getValue(PREFIX_ASSIGNMENT_RESULT).isPresent())) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAssignmentCommand.MESSAGE_USAGE));
         }
 
-        moduleName = ParserUtil.parseModuleName(argMultimap.getValue(PREFIX_NAME).get()).fullName;
-        String assignmentName = ParserUtil.parseAssignmentName(argMultimap.getValue(PREFIX_ADD_NEW_ASSIGNMENT).get());
-        double assignmentPercentage = ParserUtil.parseAssignmentPercentage(
-                argMultimap.getValue(PREFIX_PERCENTAGE_ASSIGNMENT).get());
-        double assignmentResult = ParserUtil.parseAssignmentResult(
-                argMultimap.getValue(PREFIX_RESULT_ASSIGNMENT).get());
+        moduleName = ParserUtil.parseModuleName(argMultimap.getValue(PREFIX_NAME).get());
+        try {
```

Comment:
I think there is no need for multiple try catch blocks here. We can combine all into one block since all the try catch blocks below are catching the same ParseException and handling the exception in the same way.